docs/design/services/dynamodb/high-level-library/prototype/option-1/sync/Prototype.java

```diff
@@ -0,0 +1,1026 @@
+/**
+ * The entry-point for all DynamoDB client creation. All Java Dynamo features will be accessible through this single class. This
+ * enables easy access to the different abstractions that the AWS SDK for Java provides (in exchange for a bigger JAR size).
+ *
+ * <p>
+ *     <b>Maven Module Location</b>
+ *     This would be in a separate maven module (software.amazon.awssdk:dynamodb-all) that depends on all other DynamoDB modules
+ *     (software.amazon.awssdk:dynamodb, software.amazon.awssdk:dynamodb-document). Customers that only want one specific client
+ *     could instead depend directly on the module that contains it.
+ * </p>
+ */
+@ThreadSafe
+public interface DynamoDb {
+    /**
+     * Create a low-level DynamoDB client with default configuration. Equivalent to DynamoDbClient.create().
+     * Already GA in module software.amazon.awssdk:dynamodb.
+     */
+    DynamoDbClient client();
+
+    /**
+     * Create a low-level DynamoDB client builder. Equivalent to DynamoDbClient.builder().
+     * Already GA in module software.amazon.awssdk:dynamodb.
+     */
+    DynamoDbClientBuilder clientBuilder();
+
+    /**
+     * Create a high-level "document" DynamoDB client with default configuration.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DynamoDbDocumentClient client = DynamoDb.documentClient()) {
+     *         client.listTables().tables().forEach(System.out::println);
+     *     }
+     * </code>
+     *
+     * @see DynamoDbDocumentClient
+     */
+    DynamoDbDocumentClient documentClient();
+
+    /**
+     * Create a high-level "document" DynamoDB client builder that can configure and create high-level "document" DynamoDB
+     * clients.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DynamoDbClient lowLevelClient = DynamoDb.client();
+     *          DynamoDbDocumentClient client = DynamoDb.documentClientBuilder()
+     *                                                  .dynamoDbClient(lowLevelClient)
+     *                                                  .build()) {
+     *         client.listTables().tables().forEach(System.out::println);
+     *     }
+     * </code>
+     *
+     * @see DynamoDbDocumentClient.Builder
+     */
+    DynamoDbDocumentClient.Builder documentClientBuilder();
+}
+
+/**
+ * A synchronous client for interacting with DynamoDB. While the low-level {@link DynamoDbClient} is generated from a service
+ * model, this client is hand-written and provides a richer client experience for DynamoDB.
```

Comment:
Not a design question: As the goals indicate HLL should be the preferred way, we need to hand-code every single API. I feel this might become like v1 s3 client - where we need to be up-to-date whenever new features are added in DynamoDB. Are there any parts which we can generate? (its an implementation detail, so we can discuss later too)

Reply:
It would be nice to generate certain parts or reuse the low-level abstractions, where it makes sense. I do this in a few places (or I did - not sure if I removed it), where we expose the low-level put response on the high-level putObject/putItem results. Hopefully we can find more opportunities like that throughout the implementation.